### ISSUE TITLE
[📚] Confusion in iOS QuickStart, resulting in spurious error message

### ISSUE THREAD

**Issue by jimmoy** [invertase/react-native-firebase#5739]
## Documentation Feedback

The QuickStart for iOS indicates this should be incorporated into the AppDelegate's didFinishLaunchingWithOptions:
```
if ([FIRApp defaultApp] == nil) {
  [FIRApp configure];
}
```
However, when the `nil` case is true, the `defaultApp` method logs a confusing error message, because I haven't called `configure` yet:

https://github.com/firebase/firebase-ios-sdk/blob/master/FirebaseCore/Sources/FIRApp.m#L247

So I am currently initializing like this:
```
[FIRApp configure];
```
(No `if` test.) 

Thank you for your work on this library, it has made our Firebase use very convenient.

**Comment by mikehardy** [comment]
The log message is preferable to the crash you might get here https://github.com/firebase/firebase-ios-sdk/blob/5448f59ec1c0172edd7536c38362a74d653439d1/FirebaseCore/Sources/FIRApp.m#L225

So I think it was just balancing that. Under what circumstances could that be called twice? Unsure - and perhaps it is unnecessary? Current upstream docs do not have that check: https://firebase.google.com/docs/ios/setup#initialize-firebase

When I check blame on our docs - that has been there for more than 2 years and the message has been logged about the same - perhaps it is just a really really old style ?

**Comment by jimmoy** [comment]
Yup, that may very well be (preferable to the crash), I just wanted to note it in case this was an issue. I see you've got a dup issue, so I'll go ahead and close this.

If I remove the `configure()` call from AppDelegate altogether, `initializeApp()` seems to take care of calling it indirectly. Then the only issue is that the startup sequence causes the original message I reported, and since I don't normally have Xcode up it is lost. I only happened to notice it because I was upgrading from v6 and so was on the lookout for things I may have done wrong. The Firebase calls seem to run successfully in any event, so I'm good.

The `defaultApp` method will get called multiple times since I'm using different parts of the library. I set a breakpoint on it and saw it was getting hit from `messaging()`, `installations()`, `initializeApp()`.
